src/coreclr/src/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -471,7 +512,12 @@ internal Type[] GetInstantiationPublic()
 
         internal RuntimeType Instantiate(Type[]? inst)
         {
-            // defensive copy to be sure array is not mutated from the outside during processing
+            // Defensive copy to ensure array is not mutated while during processing.
```

Comment:
This defensive copy is done in RuntimeType.MakeGenericType already, together with other argument validation. I think we just need to update the comments; or fold this into MakeGenericType since this is the only caller now.

Reply:
Updated the comments. Also fixed `RuntimeHelpers.PrepareMethod` to make a defensive copy of its args, since it was the only caller I could find that still violated this contract.